support-workers/src/main/scala/com/gu/stripe/StripeService.scala

```diff
@@ -15,16 +17,37 @@ class StripeService(config: StripeConfig, client: FutureHttpClient, baseUrl: Str
   val wsUrl = baseUrl
   val httpClient: FutureHttpClient = client
 
-  def createCustomer(token: String, currency: Currency): Future[Customer] =
-    postForm[Customer]("customers", Map(
-      "card" -> Seq(token)
-    ), getHeaders(currency))
+  def withCurrency(currency: Currency): WithCurrency = new WithCurrency(currency)
```

Comment:
I could have made this less clunky perhaps
having a class with a parameter is basically partially applying all the methods inside with whatever the parameter is.
I could have just kept the currency as a parameter to `post` and then partially applied it, rather than making a class and wrapping everything.

The ultimate aim was I just wanted to get away from passing currency and stripeService all the way through a load of methods higher up, only in order to combine them at the bottom anyway.

Reply:
How about:
`case class StripeServiceWithCurrency(service: StripeService, currency: Currency)`

Reply:
I think this is less clear than what we had before, see comment below